src/rgw/rgw_asio_client.cc

```diff
@@ -20,7 +20,7 @@ ClientIO::ClientIO(tcp::socket& socket,
 
 ClientIO::~ClientIO() = default;
 
-void ClientIO::init_env(CephContext *cct)
+int ClientIO::init_env(CephContext *cct)
```

Comment:
Consider the nodiscard attribute if this is intended for error detection.

Reply:
yeah this particular code block doesn't return an error right now, thought I don't mind adding